src/deepsparse/pipeline.py

```diff
@@ -397,6 +403,17 @@ def validate(self):
         elif isinstance(router_validation, str):
             raise ValueError(f"Invalid Router for operators: {router_validation}")
 
+    def run_func_with_middleware(
```

Comment:
can we just update `run_func` to have an optional `middleware_manager` argument?

Reply:
I thought about that but
1. this func is outside of pipeline - ideally want to keep the middleware logic in Pipeline
2. need to mutate Operator in kwarg. The signature is
```python3
def run_func(
    *args,
    func: Callable,
    inp: Any = None,
    **kwargs,
):
```

Reply:
is there anyway to combine them? seems like we have too many chain of run functions now

Reply:
agreed, middleware is used specifically to wrap a callable, so I don't think it's out of scope to add as an optional feature of run func. the pipeline still owns management of the manager, this is just to simplify the callstack within pipeline